### PR TITLE
[TD] Add support for demo data

### PR DESCRIPTION
**Pull request by hifi**

### CODE REVIEW

**Comment by hifi** [comment]
There are still `ifdef DEMO` stuff around that were mostly to restrict the engine rather than make it behave differently, need to clean those up later when we are certain they can be removed.

This together with some DOS mode fixes/updates should be quite accuarate with correct movies and nag screens.

EDIT: Oh, and I fixed palette fade on SDL with TD.

EDIT2: And this also works with or without `DEMOM.MIX` and behaves correctly. I've only tested the FTP demo (`cc1demo1.zip` + `cc1demo2.zip`)

**Comment by giulianobelinassi** [comment]
For some reason, the DEMO crashes with an OUT-OF-MEMORY error when launching the second mission.

File: tiberiandawn/init.cpp [hunk 1]
```diff
@@ -1573,120 +1519,112 @@ static void Play_Intro(bool for_real)
...
-        "TBRINFO2",
-        "TBRINFO3",
-        "TIBERFX",
-        "TRTKIL_D",
-        "TURTKILL",
-        "VISOR",
-//#endif
-#endif
-        NULL};
+    static char* _names[] = {"INTRO2",
```
Comment (giulianobelinassi) [inline tiberiandawn/init.cpp]: static const char*

**Review by giulianobelinassi — APPROVED** [review]
I will approve this, however, a bug report must be open on merge stating that it can't get past mission 2

**Comment by OmniBlade** [comment]
I can confirm I also see the crash after leaving the score screen following the first mission. Score screen doesn't update correctly either, not sure if that is a bug with the 320x200 mode or using the demo data.

**Comment by giulianobelinassi** [comment]
> I can confirm I also see the crash after leaving the score screen following the first mission. Score screen doesn't update correctly either, not sure if that is a bug with the 320x200 mode or using the demo data.

The scorescreen update is  a bug on 320x200 mode.

**Comment by hifi** [comment]
Fixed the mission endings. Seems to work now except score screen but thats not related to demo like said.

**Review by OmniBlade — APPROVED** [review]
I noticed an issue with text not drawing in tool tips, but agian not sure if that is just a bug with dos mode in general or with the demo data.

**Comment by giulianobelinassi** [comment]
> I noticed an issue with text not drawing in tool tips, but agian not sure if that is just a bug with dos mode in general or with the demo data.

Works fine here.

**Comment by hifi** [comment]
I've had that issue with DOS mode since it was merged.

**Pull request merged**